bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/PrecomputeAPI.cs

```diff
@@ -174,35 +208,17 @@ public void StopPrecompute()
         /// </summary>
         private void WorkerMethod()
         {
-            double progress = 0;
-            if (cancelTokenSource == null)
-            {
-                cancelTokenSource = new CancellationTokenSource();
-            }
-            while (!cancelToken.IsCancellationRequested)
-            {
-                Thread.Sleep((int)delay * 1000);
-
-                if (cancelToken.IsCancellationRequested)
-                {
-                    currentStatus.CurrentState = PrecomputeState.UserStopped;
-                    OnCompletedEvent();
-                    return;
-                }
-
-                progress += 10;
-                currentStatus.Percentage = progress;
-                if (progress == 100)
-                {
-                    currentStatus.CurrentState = PrecomputeState.Completed;
-                    OnCompletedEvent();
-                    return;
-                }
-                else
-                {
-                    OnProgressEvent();
-                }
-            }
+            NativePrecomputeBuffers.Populate(elgamalPublicKey.Handle, max_buffers);
+
+            int count = -1;
+            int queue_size = -2;
+            GetProgress(out count, out queue_size);
+            if (count == queue_size)
+                currentStatus.CurrentState = PrecomputeState.Completed;
+            currentStatus.Percentage = (double)count / queue_size;

```

Comment:
Possible divide by zero error?

Reply:
queue_size is defaulting to 5000, and if the user sets it to 0 then the default is used.  There should never be a divide by zero at this stage.